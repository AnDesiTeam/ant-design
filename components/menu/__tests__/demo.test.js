import demoTest from '../../../tests/shared/demoTest';

demoTest('menu', {skip: ['icon-menu-item']});
