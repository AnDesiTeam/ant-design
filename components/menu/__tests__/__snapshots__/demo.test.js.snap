// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/menu/demo/horizontal.md correctly 1`] = `
<ul
  class="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
  role="menu"
>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="display:none"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-item ant-menu-item-selected"
    role="menuitem"
  >
    <span
      class="anticon"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        fill="currentColor"
        focusable="false"
        height="1em"
        width="1em"
      >
        <use
          href="#icon-mail-o"
        />
      </svg>
    </span>
    Navigation One
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="display:none"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    aria-disabled="true"
    class="ant-menu-item ant-menu-item-disabled"
    role="menuitem"
  >
    <span
      class="anticon"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        fill="currentColor"
        focusable="false"
        height="1em"
        width="1em"
      >
        <use
          href="#icon-appstore-o"
        />
      </svg>
    </span>
    Navigation Two
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="display:none"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span
        class="submenu-title-wrapper"
      >
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-setting-o"
            />
          </svg>
        </span>
        Navigation Three - Submenu
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="display:none"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-item"
    role="menuitem"
  >
    <a
      href="https://ant.design"
      rel="noopener noreferrer"
      target="_blank"
    >
      Navigation Four - Link
    </a>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="visibility:hidden;position:absolute"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;

exports[`renders ./components/menu/demo/inline.md correctly 1`] = `
<ul
  class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
  role="menu"
  style="width:256px"
>
  <li
    class="ant-menu-submenu ant-menu-submenu-inline ant-menu-submenu-open ant-menu-submenu-selected"
    role="menuitem"
  >
    <div
      aria-expanded="true"
      aria-haspopup="true"
      aria-owns="sub1$Menu"
      class="ant-menu-submenu-title"
      style="padding-left:24px"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-mail-o"
            />
          </svg>
        </span>
        <span>
          Navigation One
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ant-menu  ant-menu-sub ant-menu-inline"
      id="sub1$Menu"
      role="menu"
    >
      <li
        class=" ant-menu-item-group"
      >
        <div
          class="ant-menu-item-group-title"
          title="Item 1"
        >
          Item 1
        </div>
        <ul
          class="ant-menu-item-group-list"
        >
          <li
            class="ant-menu-item ant-menu-item-selected"
            role="menuitem"
            style="padding-left:48px"
          >
            Option 1
          </li>
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left:48px"
          >
            Option 2
          </li>
        </ul>
      </li>
      <li
        class=" ant-menu-item-group"
      >
        <div
          class="ant-menu-item-group-title"
          title="Item 2"
        >
          Item 2
        </div>
        <ul
          class="ant-menu-item-group-list"
        >
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left:48px"
          >
            Option 3
          </li>
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left:48px"
          >
            Option 4
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-inline"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
      style="padding-left:24px"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-appstore-o"
            />
          </svg>
        </span>
        <span>
          Navigation Two
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
    <div />
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-inline"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
      style="padding-left:24px"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-setting-o"
            />
          </svg>
        </span>
        <span>
          Navigation Three
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
    <div />
  </li>
</ul>
`;

exports[`renders ./components/menu/demo/inline-collapsed.md correctly 1`] = `
<div
  style="width:256px"
>
  <button
    class="ant-btn ant-btn-primary"
    style="margin-bottom:16px"
    type="button"
  >
    <span
      class="anticon"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        fill="currentColor"
        focusable="false"
        height="1em"
        width="1em"
      >
        <use
          href="#icon-menu-fold-o"
        />
      </svg>
    </span>
  </button>
  <ul
    class="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
    role="menu"
  >
    <li
      class="ant-menu-item ant-menu-item-selected"
      role="menuitem"
      style="padding-left:24px"
    >
      <span
        class="anticon"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          <use
            href="#icon-pie-chart-o"
          />
        </svg>
      </span>
      <span>
        Option 1
      </span>
    </li>
    <li
      class="ant-menu-item"
      role="menuitem"
      style="padding-left:24px"
    >
      <span
        class="anticon"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          <use
            href="#icon-desktop-o"
          />
        </svg>
      </span>
      <span>
        Option 2
      </span>
    </li>
    <li
      class="ant-menu-item"
      role="menuitem"
      style="padding-left:24px"
    >
      <span
        class="anticon"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          <use
            href="#icon-inbox-o"
          />
        </svg>
      </span>
      <span>
        Option 3
      </span>
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-inline ant-menu-submenu-open"
      role="menuitem"
    >
      <div
        aria-expanded="true"
        aria-haspopup="true"
        aria-owns="sub1$Menu"
        class="ant-menu-submenu-title"
        style="padding-left:24px"
      >
        <span>
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-mail-o"
              />
            </svg>
          </span>
          <span>
            Navigation One
          </span>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
      <ul
        class="ant-menu  ant-menu-sub ant-menu-inline"
        id="sub1$Menu"
        role="menu"
      >
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 5
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 6
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 7
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 8
        </li>
      </ul>
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-inline"
      role="menuitem"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        style="padding-left:24px"
      >
        <span>
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-appstore-o"
              />
            </svg>
          </span>
          <span>
            Navigation Two
          </span>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
      <div />
    </li>
  </ul>
</div>
`;

exports[`renders ./components/menu/demo/sider-current.md correctly 1`] = `
<ul
  class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
  role="menu"
  style="width:256px"
>
  <li
    class="ant-menu-submenu ant-menu-submenu-inline ant-menu-submenu-open"
    role="menuitem"
  >
    <div
      aria-expanded="true"
      aria-haspopup="true"
      aria-owns="sub1$Menu"
      class="ant-menu-submenu-title"
      style="padding-left:24px"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-mail-o"
            />
          </svg>
        </span>
        <span>
          Navigation One
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ant-menu  ant-menu-sub ant-menu-inline"
      id="sub1$Menu"
      role="menu"
    >
      <li
        class="ant-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        Option 1
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        Option 2
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        Option 3
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        Option 4
      </li>
    </ul>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-inline"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
      style="padding-left:24px"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-appstore-o"
            />
          </svg>
        </span>
        <span>
          Navigation Two
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
    <div />
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-inline"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
      style="padding-left:24px"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-setting-o"
            />
          </svg>
        </span>
        <span>
          Navigation Three
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
    <div />
  </li>
</ul>
`;

exports[`renders ./components/menu/demo/switch-mode.md correctly 1`] = `
<div>
  <button
    aria-checked="false"
    class="ant-switch"
    role="switch"
    type="button"
  >
    <span
      class="ant-switch-inner"
    />
  </button>
   Change Mode
  <span
    class="ant-divider"
    style="margin:0 1em"
  />
  <button
    aria-checked="false"
    class="ant-switch"
    role="switch"
    type="button"
  >
    <span
      class="ant-switch-inner"
    />
  </button>
   Change Theme
  <br />
  <br />
  <ul
    class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
    role="menu"
    style="width:256px"
  >
    <li
      class="ant-menu-item ant-menu-item-selected"
      role="menuitem"
      style="padding-left:24px"
    >
      <span
        class="anticon"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          <use
            href="#icon-mail-o"
          />
        </svg>
      </span>
      Navigation One
    </li>
    <li
      class="ant-menu-item"
      role="menuitem"
      style="padding-left:24px"
    >
      <span
        class="anticon"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          <use
            href="#icon-calendar-o"
          />
        </svg>
      </span>
      Navigation Two
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-inline ant-menu-submenu-open"
      role="menuitem"
    >
      <div
        aria-expanded="true"
        aria-haspopup="true"
        aria-owns="sub1$Menu"
        class="ant-menu-submenu-title"
        style="padding-left:24px"
      >
        <span>
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-appstore-o"
              />
            </svg>
          </span>
          <span>
            Navigation Three
          </span>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
      <ul
        class="ant-menu  ant-menu-sub ant-menu-inline"
        id="sub1$Menu"
        role="menu"
      >
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 3
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 4
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-inline"
          role="menuitem"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            style="padding-left:48px"
            title="Submenu"
          >
            Submenu
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
          <div />
        </li>
      </ul>
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-inline"
      role="menuitem"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        style="padding-left:24px"
      >
        <span>
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-setting-o"
              />
            </svg>
          </span>
          <span>
            Navigation Four
          </span>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
      <div />
    </li>
  </ul>
</div>
`;

exports[`renders ./components/menu/demo/theme.md correctly 1`] = `
<div>
  <button
    aria-checked="true"
    checked=""
    class="ant-switch ant-switch-checked"
    role="switch"
    type="button"
  >
    <span
      class="ant-switch-inner"
    >
      Dark
    </span>
  </button>
  <br />
  <br />
  <ul
    class="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
    role="menu"
    style="width:256px"
  >
    <li
      class="ant-menu-submenu ant-menu-submenu-inline ant-menu-submenu-open ant-menu-submenu-selected"
      role="menuitem"
    >
      <div
        aria-expanded="true"
        aria-haspopup="true"
        aria-owns="sub1$Menu"
        class="ant-menu-submenu-title"
        style="padding-left:24px"
      >
        <span>
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-mail-o"
              />
            </svg>
          </span>
          <span>
            Navigation One
          </span>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
      <ul
        class="ant-menu  ant-menu-sub ant-menu-inline"
        id="sub1$Menu"
        role="menu"
      >
        <li
          class="ant-menu-item ant-menu-item-selected"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 1
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 2
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 3
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:48px"
        >
          Option 4
        </li>
      </ul>
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-inline"
      role="menuitem"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        style="padding-left:24px"
      >
        <span>
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-appstore-o"
              />
            </svg>
          </span>
          <span>
            Navigtion Two
          </span>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
      <div />
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-inline"
      role="menuitem"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        style="padding-left:24px"
      >
        <span>
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-setting-o"
              />
            </svg>
          </span>
          <span>
            Navigation Three
          </span>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
      <div />
    </li>
  </ul>
</div>
`;

exports[`renders ./components/menu/demo/vertical.md correctly 1`] = `
<ul
  class="ant-menu ant-menu-light ant-menu-root ant-menu-vertical"
  role="menu"
  style="width:256px"
>
  <li
    class="ant-menu-submenu ant-menu-submenu-vertical"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-mail-o"
            />
          </svg>
        </span>
        <span>
          Navigation One
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-vertical"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-appstore-o"
            />
          </svg>
        </span>
        <span>
          Navigation Two
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-vertical"
    role="menuitem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
    >
      <span>
        <span
          class="anticon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            <use
              href="#icon-setting-o"
            />
          </svg>
        </span>
        <span>
          Navigation Three
        </span>
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;
